include/xsimd/math/xsimd_scalar.hpp

```diff
@@ -0,0 +1,292 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XSIMD_SCALAR_HPP
+#define XSIMD_SCALAR_HPP
+
+#include <cmath>
+
+namespace xsimd {
+    /*********************************************
+     * scalar fallback for xsimd math operations *
+     *********************************************/
+
+    using std::abs;
+    using std::acos;
+    using std::acosh;
+    using std::asin;
+    using std::asinh;
+    using std::atan2;
+    using std::atan;
+    using std::atanh;
+    using std::cbrt;
+    using std::ceil;
+    using std::copysign;
+    using std::cos;
+    using std::cosh;
+    using std::erf;
+    using std::erfc;
+    using std::exp2;
+    using std::exp;
+    using std::expm1;
+    using std::fdim;
+    using std::fmax;
+    using std::fmin;
+    using std::floor;
+    using std::fmod;
+    using std::hypot;
+    using std::isfinite;
+    using std::isinf;
+    using std::lgamma;
+    using std::ldexp;
+    using std::log10;
+    using std::log1p;
+    using std::log2;
+    using std::log;
+    using std::nearbyint;
+    using std::proj;
+    using std::remainder;
+    using std::rint;
+    using std::rint;
+    using std::round;
+    using std::round;
+    using std::sin;
+    using std::sinh;
+    using std::sqrt;
+    using std::tan;
+    using std::tanh;
+    using std::tgamma;
+    using std::trunc;
+    using std::trunc;
+
+#ifdef XSIMD_ENABLE_XTL_COMPLEX
+    using xtl::abs;
+    using xtl::norm;
+    using xtl::proj;
+    using xtl::exp;
+    using xtl::log;
+    using xtl::log10;
+    using xtl::pow;
+    using xtl::sqrt;
+    using xtl::sin;
+    using xtl::cos;
+    using xtl::tan;
+    using xtl::asin;
+    using xtl::acos;
+    using xtl::atan;
+    using xtl::sinh;
+    using xtl::cosh;
+    using xtl::tanh;
+    using xtl::asinh;
+    using xtl::acosh;
+    using xtl::atanh;
+#endif
+
+    template <class T>
+    inline bool is_flint(const T& x)
+    {
+        return std::isnan(x - x) ? std::numeric_limits<T>::NaN() : x - std::trunc(x);
```

Comment:
weird, I can't find the docs for `std::numeric_limits<T>::NaN()` — only `quiet_NaN` and `signaling_NaN`. 

Is this code tested or am I missing something?

Otherwise, should we use a signalling vs quiet NaN? I think in other places we use quiet NaN's...

Reply:
after checking in cling, it seems to actually not exist.

Reply:
fixed

Reply:
I used your test suite as a base, ``is_flint`` appears to be untested :-/

Reply:
it might be that is_flint is "implicitly" tested as part of some e.g. trigo routine -- but yeah, there might be some test cases missing.